### ISSUE TITLE
Prefect 2 not compatible with click 8.1.0

### ISSUE THREAD

**Issue by tibuch** [PrefectHQ/prefect#5609]
## Description

Installing Prefect in a fresh conda environment and calling `prefect version` yields:
```
Traceback (most recent call last):
  File "/home/tibuch/Apps/miniconda3/envs/prefect_2/bin/prefect", line 5, in <module>
    from prefect.cli import app
  File "/home/tibuch/Apps/miniconda3/envs/prefect_2/lib/python3.7/site-packages/prefect/cli/__init__.py", line 1, in <module>
    from prefect.cli.base import app
  File "/home/tibuch/Apps/miniconda3/envs/prefect_2/lib/python3.7/site-packages/prefect/cli/base.py", line 14, in <module>
    import typer
  File "/home/tibuch/Apps/miniconda3/envs/prefect_2/lib/python3.7/site-packages/typer/__init__.py", line 12, in <module>
    from click.termui import get_terminal_size as get_terminal_size
ImportError: cannot import name 'get_terminal_size' from 'click.termui' (/home/tibuch/Apps/miniconda3/envs/prefect_2/lib/python3.7/site-packages/click/termui.py)
```

By default `click 8.1.0` is installed:
```
> conda list | grep click
click                     8.1.0                    pypi_0    pypi
```

Downgrading to `click==8.0.4` resolves the issue.

## Expected Behavior

```
Version:             2.0b2
API version:         0.3.0
Python version:      3.7.13
Git commit:          b2a048c2
Built:               Thu, Mar 17, 2022 2:24 PM
OS/Arch:             linux/x86_64
Profile:             default
Server type:         ephemeral
Server:
  Database:          sqlite
  SQLite version:    3.38.0
```

## Reproduction

```
conda create --name prefect_2 python=3.7
pip install prefect==2.0b2
prefect version
```

**Comment by madkinsz** [comment]
Thanks for the report, this is an issue in our dependency. You can follow the upstream issue at https://github.com/tiangolo/typer/issues/377. We've even contributed the fix https://github.com/tiangolo/typer/pull/375

**Comment by tiangolo** [comment]
Hello! I just released Typer `0.4.1`, thanks for the work @madkinsz! Upgrading Typer should handle it. :rocket: :nerd_face:
